ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -102,6 +106,95 @@ def _get_table_description(schema: str, table: str, inspector: Inspector) -> str
     return description
 
 
+def get_dbt_local(config):
+    if config.dbtConfig.dbtCatalogFilePath is not None:
+        with open(
+            config.dbtConfig.dbtCatalogFilePath, "r", encoding="utf-8"
+        ) as catalog:
+            dbt_catalog = catalog.read()
+    if config.dbtConfig.dbtManifestFilePath is not None:
+        with open(
+            config.dbtConfig.dbtManifestFilePath, "r", encoding="utf-8"
+        ) as manifest:
+            dbt_manifest = manifest.read()
+    return json.loads(dbt_catalog), json.loads(dbt_manifest)
+
+
+def get_dbt_http(config):
+    import urllib.request
+
+    catalog_file = urllib.request.urlopen(config.dbtConfig.dbtCatalogFilePath)
+    manifest_file = urllib.request.urlopen(config.dbtConfig.dbtManifestFilePath)
+    dbt_catalog = catalog_file.read().decode()
+    dbt_manifest = manifest_file.read().decode()
+    return json.loads(dbt_catalog), json.loads(dbt_manifest)
+
+
+def get_dbt_gcs(config):
+    dbt_options = config.dbtConfig.gcsConfig
+    if config.dbtProvider.value == "gcs":
+        options = {
+            "credentials": {
+                "type": dbt_options.type,
+                "project_id": dbt_options.projectId,
+                "private_key_id": dbt_options.privateKeyId,
+                "private_key": dbt_options.privateKey,
+                "client_email": dbt_options.clientEmail,
+                "client_id": dbt_options.clientId,
+                "auth_uri": dbt_options.authUri,
+                "token_uri": dbt_options.tokenUri,
+                "auth_provider_x509_cert_url": dbt_options.authProviderX509CertUrl,
+                "client_x509_cert_url": dbt_options.clientX509CertUrl,
+            }
+        }
+    else:
+        options = {"credentials_path": dbt_options.__root__}
+    if store_gcs_credentials(options):
+        from google.cloud import storage
+
+        client = storage.Client()
+        for bucket in client.list_buckets():
+            for blob in client.list_blobs(bucket.name):
+                if MANIFEST_FILE_NAME in blob.name:
+                    dbt_manifest = blob.download_as_string().decode()
+                if CATALOG_FILE_NAME in blob.name:
+                    dbt_catalog = blob.download_as_string().decode()
+    return json.loads(dbt_catalog), json.loads(dbt_manifest)
+
+
+def get_dbt_s3(config):
+    from metadata.utils.aws_client import AWSClient
+
+    aws_client = AWSClient(config.dbtConfig).get_resource("s3")
+    buckets = aws_client.buckets.all()
+    for bucket in buckets:
+        for bucket_object in bucket.objects.all():
+            if MANIFEST_FILE_NAME in bucket_object.key:
+                dbt_manifest = bucket_object.get()["Body"].read().decode()
+            if CATALOG_FILE_NAME in bucket_object.key:
+                dbt_catalog = bucket_object.get()["Body"].read().decode()
+    return json.loads(dbt_catalog), json.loads(dbt_manifest)
```

Comment:
same as before. Let's add validations that we can read the file and either return `json.loads(dbt_catalog)` or None properly. Here if we do not find the file it will fail with `using a variable before assignment` error as the dbt_catalog file won't be created.

One snippet I like to use for this "find in list" requirements is the following: https://snappify.io/view/66cffd5d-fa98-4e4c-8d14-acd8396eb781